pkg/registry/pod/etcd/etcd_test.go

```diff
@@ -130,6 +133,70 @@ func TestDelete(t *testing.T) {
 	test.TestDeleteGraceful(scheduledPod, 30)
 }
 
+type FailDeletionStorage struct {
+	storage.Interface
+	Called *bool
+}
+
+func (f FailDeletionStorage) Delete(ctx context.Context, key string, out runtime.Object) error {
+	*f.Called = true
+	return etcd.Error{Code: etcd.ErrorCodeKeyNotFound}
+}
+
+func newFailDeleteStorage(t *testing.T, called *bool) (*REST, *etcdtesting.EtcdTestServer) {
+	etcdStorage, server := registrytest.NewEtcdStorage(t, "")
+	failDeleteStorage := FailDeletionStorage{etcdStorage, called}
+	restOptions := generic.RESTOptions{failDeleteStorage, generic.UndecoratedStorage, 3}
+	storage := NewStorage(restOptions, nil, nil)
+	return storage.Pod, server
+}
+
+func TestIgnoreDeleteNotFound(t *testing.T) {
+	pod := validNewPod()
+	testContext := api.WithNamespace(api.NewContext(), api.NamespaceDefault)
+	called := false
+	registry, server := newFailDeleteStorage(t, &called)
+	defer server.Terminate(t)
+
+	// should fail if pod A is not created yet.
+	_, err := registry.Delete(testContext, pod.Name, nil)
+	if !errors.IsNotFound(err) {
+		t.Errorf("Unexpected error: %v", err)
+	}
+
```

Comment:
LGTM, although this should probably eventually move to rest/test.  Doesn't have to happen here.